test/ResourceMatcher.test.js

```diff
@@ -598,6 +598,157 @@ describe("testResourceMatcher", () => {
+
+        const result = rule.matchString({source: resource.source, target: resource.target, resource, file: resource.pathName});
+
+        expect(result).toBeUndefined();
+    });
+
+    // TODO @NatK: Fix in the next PR so that the rule does not run for falsy target values.
+    // it.each([
+    //     // {name: "empty", target: ""},
+    //     // {name: "undefined", target: undefined},
+    //     // {name: "undefined", target: null},
+    // ])("does not apply rule if target string is $name", ({target}) => {
+    //     const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
+    //     const resource = createTestResourceString({source: "snake_case", target});
+    //
+    //     const result = rule.matchString({source: resource.source, target: resource.target, resource, file: resource.pathName});
+    //
+    //     expect(result).toBeUndefined();
+    // });
+
+    test.each(
+        [
+            {name: "whitespace (solely)", source: " "},
+            {name: "text and whitespace", source: "snake case"},
+            {name: "snake case and text", source: "snake_case and text"},
+            {name: "screaming snake case and text", source: "SCREAMING_SNAKE_CASE and text"},
+            {name: "mixed case", source: "mixed_CASE"},
+        ]
+    )("does not apply if source string is $name", ({name, source}) => {
+        const rule = new ResourceMatcher(findRuleDefinition("resource-snake-case"));
+        const resource = createTestResourceString({source, target: "does not matter"});
+
+        const result = rule.matchString({source: resource.source, target: resource.target, resource, file: resource.pathName});
+
+        expect(result).toBeUndefined();
+    });
+
+    test.each(
+        [
+            {name: "snake case", source: "snake_case"},
+            {name: "snake case with leading and trailing whitespace", source: " snake_case "},
+            {name: "snake case with numbers (123)", source: " snake_case123 "},
+            {name: "snake case with underscored numbers (_123)", source: " snake_case_123 "},
+
+            {name: "screaming snake case", source: "SOME_SCREAMING_SNAKE_CASE"},
+            {name: "screaming snake case with leading and trailing whitespace", source: " SOME_SCREAMING_SNAKE_CASE "},
+            {name: "screaming snake case with numbers", source: "SOME_SCREAMING_SNAKE_CASE123 "},
+            {name: "screaming snake case with underscored numbers", source: "SOME_SCREAMING_SNAKE_CASE_123 "},
+
+            {name: "camel snake case", source: "camel_Snake_Case"},
+            {name: "came snake case with leading and trailing whitespace", source: " camel_Snake_Case "},
+            {name: "camel snake case with numbers", source: "camel_Snake_Case123 "},
+            {name: "camel snake case with underscored numbers", source: "camel_Snake_Case_123 "},
+            {name: "camel snake case with underscored numbers", source: "camel_Snake_Case_123 "},
+        ]
```

Comment:
Also add tests to this list:

```
"_test",      // leading underscore
"_test_test",    // leading underscore and more
"sNaKe_CaSe",    // snake case + ransom note case combined!

Reply:
Why would we treat `_leading` undersocre as valid snake case and `trailing_` as invalid one?

Reply:
Good point. Perhaps they should both be considered snake case.

Reply:
Done.